### PR TITLE
[F4 - Input parameters]

### PR DESCRIPTION
**Pull request by jamiecdavidson**
**Description**

1. Work performed on backend for API controller, routes and lib.
2. Frontend setup to take Input parameter and POST to backend.
3. Merges complete with main to update current branch with updated main.

**Trello**

- Link: https://trello.com/c/6ywnX3hQ/59-f4-input-parameters-5

### CODE REVIEW

File: backend/lib/AIArtGenerator.js [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+//       .then((data) => {
+//         console.log("test", data);
+//         return data;
+//       })
+//       .catch((error) => {
+//         //   console.log(error);
+//         return error;
+//       })
+//   );
+// };
```
Comment (xuqun123) [inline backend/lib/AIArtGenerator.js:24]: Please remove these codes.

File: frontend/src/components/GenerateBar.js [hunk 2]
```diff
@@ -3,11 +3,31 @@ import Paper from '@mui/material/Paper'
 import InputBase from '@mui/material/InputBase'
 import IconButton from '@mui/material/IconButton'
 import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import axiosClient from '../lib/axiosClient'
 
 function GenerateBar() {
+  const [inputValue, setInputValue] = React.useState('')
+
+  console.log(inputValue)
```
Comment (xuqun123) [inline frontend/src/components/GenerateBar.js]: remove this line

**Review by xuqun123 — APPROVED** [review]

**Comment by xuqun123** [comment]
Please remove those unneeded codes, then feel free to merge it.

File: backend/lib/AIArtGenerator.js (hunk 1 above)
Comment (jamiecdavidson) [reply backend/lib/AIArtGenerator.js:24]: done

File: frontend/src/components/GenerateBar.js (hunk 2 above)
Comment (jamiecdavidson) [reply frontend/src/components/GenerateBar.js]: Just Line 11?

Comment (xuqun123) [reply frontend/src/components/GenerateBar.js]: Yes

Comment (xuqun123) [reply frontend/src/components/GenerateBar.js]: No worries, i updated that for you.

File: backend/.env.example [hunk 3]
```diff
@@ -1,3 +1,4 @@
 PORT=3000
 MONGODB_URL=
-SERVER_JWT_SECRET=your_jwt_secret
\ No newline at end of file
+SERVER_JWT_SECRET=your_jwt_secret
+DEEPAI_API_KEY=7fec5eb0-443c-4a9f-95f4-29dbf032e820
```
Comment (xuqun123) [inline backend/.env.example]: This key shouldn't be your own key.

**Pull request merged**